### PR TITLE
Exosuit QOL: Action Buttons

### PR DESCRIPTION
**Pull request by PollardTheDragon**
## What Does This PR Do
Adds exosuit action buttons to swap equipment more effectively, as a QOL upgrade.

Taken from #23802 , with permission from [ErdGinalD](https://github.com/ErdGinalD)

## Why It's Good For The Game
From the original PR:
Action buttons, on the other hand, despite having an Atl+click function, are more intuitive and allow you to bind modules, fighting or fixing a station more efficiently, switching between RCD or a clamp etc. A useful and cool feature, proven over time on Russian Paradise, which never once gave the question "what's it for".

## Images of changes

image

## Testing
Spawned a few armed mechs. Swapped equipment using action buttons. Shot things. Built things. 

## Changelog
:cl:
add: Action buttons for mecha equipment
/:cl:

### CODE REVIEW

**Comment by 1080pCat** [comment]
it'd be cool to see middle mouse button to rearm, but thats probably out of the scope of this pr.

File: code/game/mecha/mecha_actions.dm [hunk 1]
```diff
@@ -231,3 +236,24 @@
 	button_icon_state = "mech_damtype_[new_damtype]"
 	playsound(src, 'sound/mecha/mechmove01.ogg', 50, 1)
 	UpdateButtons()
+
+/datum/action/innate/mecha/select_module
+	name = "Hey, you shouldn't see this"
```
Comment (1080pCat) [inline code/game/mecha/mecha_actions.dm:241]:
```suggestion
	name = "Hey, you shouldn't see this please make a bug report"
```

**Review by 1080pCat — APPROVED** [review]

**Comment by PollardTheDragon** [comment]
> it'd be cool to see middle mouse button to rearm, but thats probably out of the scope of this pr.

I want to avoid doing that, as middle mouse is how the AI uses mech equipment currently (AI powers left click, mech powers middle click).

**Review by ErdGinalD — APPROVED** [review]

**Comment by PollardTheDragon** [comment]
Commenting to unstale this.

File: code/game/mecha/equipment/mecha_equipment.dm [hunk 2]
```diff
@@ -119,11 +128,20 @@
 		if(chassis.selected == src)
 			chassis.selected = null
 		update_chassis_page()
+		if(chassis.occupant)
+			remove_targeted_action()
 		chassis.log_message("[src] removed from equipment.")
 		chassis = null
 		set_ready_state(1)
```
Comment (DGamerL) [inline code/game/mecha/equipment/mecha_equipment.dm:137]:
```suggestion
```

File: code/game/mecha/equipment/mecha_equipment.dm [hunk 3]
```diff
@@ -111,6 +111,15 @@
 	if(!M.selected)
 		M.selected = src
 	update_chassis_page()
+	if(M.occupant)
+		give_targeted_action()
+
+/obj/item/mecha_parts/mecha_equipment/proc/give_targeted_action()
+	if(!selectable)
+		return
+	var/datum/action/innate/mecha/select_module/select_action = new
```
Comment (DGamerL) [inline code/game/mecha/equipment/mecha_equipment.dm:120]:
```suggestion
	var/datum/action/innate/mecha/select_module/select_action = new()
```

File: code/game/mecha/equipment/mecha_equipment.dm [hunk 4]
```diff
@@ -141,3 +159,6 @@
 /obj/item/mecha_parts/mecha_equipment/proc/log_message(message)
 	if(chassis)
 		chassis.log_message("<i>[src]:</i> [message]")
+
+/obj/item/mecha_parts/mecha_equipment/proc/self_occupant_attack()
+	return
```
Comment (DGamerL) [inline code/game/mecha/equipment/mecha_equipment.dm:164]: Why add this proc here? Futureproofing?

Comment (PollardTheDragon) [reply code/game/mecha/equipment/mecha_equipment.dm]: It was being called from mob.dm in mode(), but after testing with it and that bit taken out it did literally nothing so it's been removed now.

File: code/game/mecha/mecha_actions.dm [hunk 5]
```diff
@@ -216,3 +220,24 @@
 	button_icon_state = "mech_damtype_[new_damtype]"
 	playsound(src, 'sound/mecha/mechmove01.ogg', 50, 1)
```
Comment (Burzah) [inline code/game/mecha/mecha_actions.dm:221]:
```suggestion
	playsound(src, 'sound/mecha/mechmove01.ogg', 50, TRUE)
```

**Review by Burzah — APPROVED** [review]

File: code/modules/tgui/states.dm [hunk 6]
```diff
@@ -128,4 +128,7 @@
 /mob/living/carbon/human/shared_living_ui_distance(atom/movable/src_object, viewcheck)
 	if(HAS_TRAIT(src, TRAIT_TELEKINESIS) && (get_dist(src, src_object) <= 2))
 		return UI_INTERACTIVE
+	if(ismecha(loc))
+		if(get_dist(loc, src_object) <= 1)
+			return UI_INTERACTIVE
```
Comment (DGamerL) [inline code/modules/tgui/states.dm:133]:
```suggestion
	if(ismecha(loc) && get_dist(loc, src_object) <= 1)
		return UI_INTERACTIVE
```

**Review by DGamerL — APPROVED** [review]

**Pull request merged**